File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/DKImagePickerControllerDemoVC.m
//
//  DKImagePickerControllerDemoVC.m
//  DKImagePickerControllerDemo_OC
//
//  Created by zm on 2017/6/23.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "DKImagePickerControllerDemoVC.h"
#import "ViewController.h"
#import "DKImagePickerController.h"
#import "CustomCameraUIDelegate.h"
#import "CustomUIDelegate.h"
#import "CustomLayoutUIDelegate.h"
@interface DKImagePickerControllerDemoVC ()

@end

@implementation DKImagePickerControllerDemoVC

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

//#pragma mark - Table view data source
//
//- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
//#warning Incomplete implementation, return the number of sections
//    return 0;
//}
//
//- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
//#warning Incomplete implementation, return the number of rows
//    return 0;
//}

/*
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:<#@"reuseIdentifier"#> forIndexPath:indexPath];
    
    // Configure the cell...
    
    return cell;
}
*/

/*
// Override to support conditional editing of the table view.
- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the specified item to be editable.
    return YES;
}
*/

/*
// Override to support editing the table view.
- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        // Delete the row from the data source
        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
    } else if (editingStyle == UITableViewCellEditingStyleInsert) {
        // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
    }   
}
*/

/*
// Override to support rearranging the table view.
- (void)tableView:(UITableView *)tableView moveRowAtIndexPath:(NSIndexPath *)fromIndexPath toIndexPath:(NSIndexPath *)toIndexPath {
}
*/

/*
// Override to support conditional rearranging of the table view.
- (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the item to be re-orderable.
    return YES;
}
*/


#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
    
    UITableViewCell * cell = (UITableViewCell *)sender;
    ViewController * vc = (ViewController *)segue.destinationViewController;
    vc.title= cell.textLabel.text;
    if ([segue.identifier isEqualToString:@"Pick All"]) {
        vc.pickerController = [DKImagePickerController new];
    }
    if ([segue.identifier isEqualToString:@"Pick Photos Only"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.assetType = DKImagePickerControllerAssetAllPhotosType;
        vc.pickerController = pc;
    }
    
    if ([segue.identifier isEqualToString:@"Pick Videos Only"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.assetType = DKImagePickerControllerAssetAllVideosType;
        vc.pickerController = pc;
    }
    
    if ([segue.identifier isEqualToString:@"Pick All(Only Photos Or Videos)"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.allowMultipleTypes = NO;
        vc.pickerController = pc;
    }
    if ([segue.identifier isEqualToString:@"Single Select"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.singleSelect = YES;
        vc.pickerController = pc;

    }
    
    if ([segue.identifier isEqualToString:@"Take A Picture"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.sourceType = DKImagePickerControllerSourceCameraType;
        vc.pickerController = pc;
    }
    
    if ([segue.identifier isEqualToString:@"Hides Camera"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.sourceType = DKImagePickerControllerSourcePhotoType;
        vc.pickerController = pc;
    }
    
    if ([segue.identifier isEqualToString:@"Allows Landscape"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.allowsLandscape = YES;
        vc.pickerController = pc;
    }
    
    if ([segue.identifier isEqualToString:@"Camera Customization"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.UIDelegate = [CustomCameraUIDelegate new];
        pc.modalPresentationStyle = UIModalPresentationOverCurrentContext;
        vc.pickerController = pc;

    }
    if ([segue.identifier isEqualToString:@"UI Customization"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.UIDelegate = [CustomUIDelegate new];
        pc.showsCancelButton = YES;
        pc.modalPresentationStyle = UIModalPresentationOverCurrentContext;
        vc.pickerController = pc;
    }
    
    if ([segue.identifier isEqualToString:@"Layout Customization"]) {
        DKImagePickerController * pc  = [DKImagePickerController new];
        pc.UIDelegate = [CustomLayoutUIDelegate new];
        pc.modalPresentationStyle = UIModalPresentationOverCurrentContext;
        vc.pickerController = pc;
    }
    
    
}


@end

File: DKImagePickerControllerDemo/DKImagePickerControllerDemoVC.swift
//
//  DKImagePickerControllerDemoVC.swift
//  DKImagePickerControllerDemo
//
//  Created by ZhangAo on 03/01/2017.
//  Copyright © 2017 ZhangAo. All rights reserved.
//

import UIKit

class DKImagePickerControllerDemoVC: UITableViewController {
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        let cell = sender as! UITableViewCell
        
        let destination = segue.destination as! ViewController
        destination.title = cell.textLabel?.text
        
        switch segue.identifier! {
            
        case "Pick All":
            let pickerController = DKImagePickerController()
            
            destination.pickerController = pickerController
            
        case "Pick Photos Only":
            let pickerController = DKImagePickerController()
            pickerController.assetType = .allPhotos
            
            destination.pickerController = pickerController
            
        case "Pick Videos Only":
            let pickerController = DKImagePickerController()
            pickerController.assetType = .allVideos
            
            destination.pickerController = pickerController
            
        case "Pick All(Only Photos Or Videos)":
            let pickerController = DKImagePickerController()
            pickerController.allowMultipleTypes = false
            
            destination.pickerController = pickerController
            
        case "Take A Picture":
            let pickerController = DKImagePickerController()
            pickerController.sourceType = .camera
            
            destination.pickerController = pickerController
            
        case "Hides Camera":
            let pickerController = DKImagePickerController()
            pickerController.sourceType = .photo
            
            destination.pickerController = pickerController
            
        case "Allows Landscape":
            let pickerController = DKImagePickerController()
            pickerController.allowsLandscape = true
            
            destination.pickerController = pickerController
            
        case "Single Select":
            let pickerController = DKImagePickerController()
            pickerController.singleSelect = true
            pickerController.autoCloseOnSingleSelect = true
            
            destination.pickerController = pickerController
            
        case "Camera Customization":
            let pickerController = DKImagePickerController()
            pickerController.UIDelegate = CustomCameraUIDelegate()
            pickerController.modalPresentationStyle = .overCurrentContext
            
            destination.pickerController = pickerController
            
        case "UI Customization":
            let pickerController = DKImagePickerController()
            pickerController.sourceType = .photo
            pickerController.UIDelegate = CustomUIDelegate()
            pickerController.showsCancelButton = true
            
            destination.pickerController = pickerController
            
        case "Layout Customization":
            let pickerController = DKImagePickerController()
            pickerController.UIDelegate = CustomLayoutUIDelegate()
            
            destination.pickerController = pickerController
            
        case "Inline":
            let pickerController = DKImagePickerController()
            pickerController.inline = true
            pickerController.fetchLimit = 10
            pickerController.UIDelegate = CustomInlineLayoutUIDelegate()
            pickerController.assetType = .allPhotos
            pickerController.sourceType = .photo
            
            destination.pickerController = pickerController
            
        default:
            assert(false)
        }
    }
}
